src/list-factory.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+var ResponseHandler = require('./request-handler');
+
+function ListFactory(retriever, spec) {
```

Comment:
why `spec` here and `options` elsewhere? is there a difference you're trying to highlight?